src/pudl/metadata/classes.py

```diff
@@ -1292,6 +1291,22 @@ class Resource(PudlMeta):
         "contributors", "keywords", "licenses", "sources", fn=_check_unique
     )
 
+    @property
+    def sphinx_ref_name(self):
```

Comment:
This basically moves the logic of the `format_resource_name_cross_ref` jinja filter to `Resource`.